who_l3_smart_tools/core/terminology/terminology.py

```diff
@@ -0,0 +1,256 @@
+        row (dict): The row data.
+        schema (type[ConceptSchema]): The schema for the concept.
+        **kwargs: Additional keyword arguments.
+
+    Attributes:
+        row (dict): The row data.
+        schema (type[ConceptSchema]): The schema for the concept.
+        converted_row (dict): The processed and converted row data.
+
+    Methods:
+        _process_row_required_fields: Processes the required fields of the row.
+        process_for_csv: Processes the row for CSV output.
+        process_for_json: Processes the row for JSON output.
+    """
+
+    def __init__(
+        self,
+        row: dict,
+        schema: type[ConceptSchema],
+        concept_class: str,
+        values_set: Union[dict, None],
+        **kwargs,
+    ) -> None:
+        self.row = row
+        self.schema = schema
+        self.concept_class = concept_class
+        self.converted_row = None
+        self.value_set = values_set
+        self._process_row_required_fields(kwargs)
+        self._process_value_set()
+
+    def _process_row_required_fields(self, extra_args: dict):
+        """
+        Processes the required fields of the row.
+
+        Args:
+            extra_args (dict): Additional arguments.
+
+        Returns:
+            dict: The processed row data.
+        """
+        converted_row = {
+            "id": self.row.pop(self.schema.id),
+            "name": self.row.pop(self.schema.name),
+            "datatype": self.row.pop(self.schema.datatype),
+            "description": self.row.pop(self.schema.description),
+            "concept_class": self.concept_class,
+        }
+        for index, name in enumerate(self.schema.additional_names):
+            converted_row[f"name[{index+1}]"] = self.row.pop(name)
+        for index, description in enumerate(self.schema.additional_descriptions):
+            converted_row[f"description[{index+1}]"] = self.row.pop(description)
+        converted_row.update(extra_args)
+        self.converted_row = converted_row
+
+    def _process_value_set(self):
+        if self.converted_row.get("datatype").lower() == "codes":
+            if not self.value_set:
+                raise ValueError(f"Value set is required for datatype 'codes'. {self.converted_row}")
+            codings = self.value_set["type"]["coding"]
+            codings.append(
+                {
+                    "system": "http://hl7.org/fhir/v2/0203",
+                    "code": self.converted_row["id"].replace(".", ""),
+                    "display": self.converted_row["name"],
+                }
+            )
+        elif self.converted_row.get("datatype").lower() == "coding":
+            self.value_set = {
+                "system": "https://fhir.staging.aws.openconceptlab.org",
+                "type": {"coding": [], "text": self.converted_row["name"]},
+                "value": self.converted_row["id"].replace(".", ""),
+            }
+        else:
+            self.value_set = None
+
+    def process_for_csv(self):
```

Comment:
IMO, rather than operating on a shared object (`self.converted_row()`), this should probably just return the row value in the expected format.